design/BufferOperations.md

```diff
@@ -181,3 +148,111 @@ stagingVertexBuffer.mapWriteAsync().then((stagingData) => {
     // Enqueue copy from the staging buffer to the real vertex buffer.
 });
 ```
+
+### Updating data to an existing buffer (like webgl.bufferSubData)
+
+```
+function bufferSubData(device, destBuffer, destOffset, srcArrayBuffer) {
+    const byteCount = srcArrayBuffer.byteLength;
+    const (srcBuffer, mapping) = device.createBufferMapped({
+        size: byteCount,
+        usage: GPUBufferUsage.MAP_WRITE | GPUBufferUsage.TRANSFER_SRC,
+    });
+    (new Uint8Array(mapping)).set(new Uint8Array(srcArrayBuffer)); // memcpy
+    srcBuffer.unmap();
+
+    const enc = device.createCommandEncoder({});
+    enc.copyBufferToBuffer(srcBuffer, 0, destBuffer, destOffset, byteCount);
+    const cb = enc.finish();
+    const q = device.getQueue();
+    q.submit([cb]);
+
+    srcBuffer.destroy();
+}
+```
+
+As usual, batching per-frame uploads through fewer (or a single) buffer reduces
+overhead.
+
+Applications are free to implement their own heuristics for batching or reusing
+upload buffers:
+```
+function AutoRingBuffer(device, chunkSize) {
+    const queue = device.getQueue();
+    let availChunks = [];
+
+    function Chunk() {
+        const size = chunkSize;
+        const (buf, initialMap) = this.device.createBufferMapped({
```

Comment:
I believe the way this is returned from IDL, it should be destructured as a tuple: `[buf, initialMap]`